File: fixtures/scale/csharp/medium/Ast/Assignment.cs
namespace Calc.Ast
{
    public class Assignment : Node
    {
        public string Name { get; }
        public Node Value { get; }

        public Assignment(string name, Node value)
        {
            Name = name;
            Value = value;
        }

        public override string Describe()
        {
            return Name + " = " + Value.Describe();
        }
    }
}

File: fixtures/scale/swift/medium/Ast/Assignment.swift
// A variable binding statement, e.g. "x = 1 + 2".
//
// Assigning also yields the assigned value, so "x = 1 + 2" prints "= 3".
struct Assignment: Node {
    let name: String
    let value: Node

    func describe() -> String {
        return name + " = " + value.describe()
    }
}
